spec/classes/caddy_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'spec_helper'
+
+describe 'caddy', type: :class do
+  on_supported_os(facterversion: '3.11.7').each do |os, facts|
+    context "on #{os}" do
+      architecture = facts[:architecture]
+      os_family    = facts[:os]['family']
+      os_name      = facts[:os]['name']
+      # os_distro    = facts[:os]['distro']['codename']
+      os_release   = facts[:os]['release']['major']
+
+      case "#{os_name}-#{os_release}"
```

Comment:
IMHO it's cleaner to use a custom fact. https://github.com/theforeman/puppet-puppet/blob/c6d90aba578dd5ceb34b789f57e4bd7eac6a71cc/spec/spec_helper.rb#L18-L41 is an example.

Reply:
I wanted to use the fact ```service_provider``` but it does not work as expected.